### DISCUSSION TITLE
How to conditionally compile with SPA mode?

### DISCUSSION THREAD

**Discussion by aschultz** [remix-run/remix#9504] [Q&A]
I'm working on a project where we want to be able to ship an Electron-style client app and include in the install package a pre-built version of the web app content. We'll use this content if the user goes offline and first tries to launch the app while still offline, otherwise if they're online we'll refresh and fetch fresh from the internet, using service workers to cache everything for future offline use.

Given this requirement, it seems like we'd want to have an SPA build of our app for deployment in the client package, and a non-SPA build to deploy onto our web servers.

It's easy enough to toggle the Remix SSR flag in vite.config.js. However, if any of the routes include server-only exports like loader/action/headers, the SPA build fails. Wrapping the exports in conditions, e.g. `if (!__SPA__) { export const loader = async () => {} }` doesn't seem to be supported by the language. 

What should I do here to enable the same route files to be used in both environments?

**Comment by jrestall** [comment]
What happens if you use pnpm patch <pkg> to remove [the check](https://github.com/remix-run/remix/blob/ffbfeaeff6cf2c1a00819041c3a2c8c93d5444ed/packages/remix-dev/vite/plugin.ts#L1600) from the remix vite plugin?
